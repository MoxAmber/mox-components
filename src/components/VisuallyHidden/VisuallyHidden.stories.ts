import type { Meta, StoryObj } from "@storybook/web-components";
import { html } from "lit";
import ".";
import { VisuallyHidden } from "./index.js";

interface CustomArgs extends VisuallyHidden {
  content?: string;
}

const meta = {
  component: "mox-visually-hidden",
  args: {
    content: "This text is only visible in the accessibility tree.",
  },
  render: (args) => html`
    <mox-visually-hidden ?focusable=${args.focusable}>
      <a href="#">${args.content}</a>
    </mox-visually-hidden>
  `,
} satisfies Meta<CustomArgs>;
export default meta;

export const Primary: StoryObj = {};
export const Focusable: StoryObj = { args: { focusable: true } };
