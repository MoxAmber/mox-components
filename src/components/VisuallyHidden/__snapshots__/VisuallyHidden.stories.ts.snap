// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/VisuallyHidden Focusable smoke-test: shadow DOM 1`] = `
<slot class=" focusable ">
</slot>
`;

exports[`components/VisuallyHidden Primary smoke-test: shadow DOM 1`] = `
<slot class="  ">
</slot>
`;
